### PR TITLE
ACS-6376: ingest updated metadata

### PR DESCRIPTION
**Pull request by Pawel-608**

### CODE REVIEW

File: live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/messaging/in/RepoEventMapperTest.java [hunk 1]
```diff
@@ -49,10 +50,7 @@
 import org.alfresco.hxi_connector.live_ingester.domain.usecase.metadata.IngestMetadataCommand;
 import org.alfresco.hxi_connector.live_ingester.domain.usecase.metadata.model.CustomPropertyDelta;
 import org.alfresco.hxi_connector.live_ingester.messaging.in.mapper.RepoEventMapper;
-import org.alfresco.repo.event.v1.model.DataAttributes;
-import org.alfresco.repo.event.v1.model.NodeResource;
-import org.alfresco.repo.event.v1.model.RepoEvent;
-import org.alfresco.repo.event.v1.model.UserInfo;
+import org.alfresco.repo.event.v1.model.*;
 
 class RepoEventMapperTest
```
Comment (Pawel-608) [inline live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/messaging/in/RepoEventMapperTest.java:55]: I don't like this test at all but i don't see any easy way to improve it. Do you have any idea @krdabrowski, @mpichura?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java [hunk 2]
```diff
@@ -54,39 +63,113 @@ public class RepoEventMapper
...
-                updated(event.getData().getResource().getPrimaryAssocQName()),
-                updated(event.getData().getResource().getNodeType()),
-                updated(event.getData().getResource().getCreatedByUser().getId()),
-                updated(event.getData().getResource().getModifiedByUser().getId()),
-                updated(event.getData().getResource().getAspectNames()),
-                updated(event.getData().getResource().isFile()),
-                updated(event.getData().getResource().isFolder()),
-                updated(event.getData().getResource().getCreatedAt().toInstant().toEpochMilli()),
-                allCustomPropertiesUpdated(event));
+                calculatePropertyDelta(event, NodeResource::getName),
```
Comment (krdabrowski) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java:80]: If `node` is nullable it won't fail on NPE, while trying to obtain the `name` and rest of the data?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/usecase/metadata/IngestMetadataCommand.java [hunk 3]
```diff
@@ -31,12 +31,9 @@
 
 import java.util.Set;
 
-import org.springframework.validation.annotation.Validated;
-
 import org.alfresco.hxi_connector.live_ingester.domain.usecase.metadata.model.CustomPropertyDelta;
 import org.alfresco.hxi_connector.live_ingester.domain.usecase.metadata.model.PropertyDelta;
 
-@Validated
```
Comment (krdabrowski) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/usecase/metadata/IngestMetadataCommand.java:39]: Why fields are not validated any more?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java [hunk 4]
```diff
@@ -54,39 +63,113 @@ public class RepoEventMapper
...
                 .filter(property -> Objects.nonNull(property.getValue()))
                 .map(property -> CustomPropertyDelta.updated(property.getKey(), property.getValue()))
                 .collect(Collectors.toSet());
     }
+
+    private Stream<Map.Entry<String, ?>> customPropertiesStream(NodeResource node)
+    {
+        return ofNullable(node)
+                .map(NodeResource::getProperties)
+                .map(Map::entrySet)
```
Comment (krdabrowski) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java:157]: Hmm... `.flatMap(Map::entrySet);` won't be enough?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java [hunk 5]
```diff
@@ -54,39 +63,113 @@ public class RepoEventMapper
...
+
+    private Set<CustomPropertyDelta<?>> someCustomPropertiesUpdated(RepoEvent<DataAttributes<NodeResource>> event)
+    {
+        return customPropertiesStream(event.getData().getResourceBefore())
+                .map(Map.Entry::getKey)
+                .map(changedPropertyName -> toCustomPropertyDelta(event.getData(), changedPropertyName))
+                .collect(Collectors.toSet());
+    }
+
+    private CustomPropertyDelta<?> toCustomPropertyDelta(DataAttributes<NodeResource> event, String changedPropertyName)
```
Comment (krdabrowski) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java:138]:
```suggestion
    private CustomPropertyDelta<?> toCustomPropertyDelta(DataAttributes<NodeResource> eventData, String changedPropertyName)
```

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/usecase/metadata/IngestMetadataCommand.java (hunk 3 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/domain/usecase/metadata/IngestMetadataCommand.java:39]: there is no validation with use of annotations so it is unnecessary - this annotation was left by my by mistake after testing jakarta validation annotations

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java (hunk 2 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java:80]: `nodeResourceBefore` is nullable but later on we are wrapping it with an optional to ensure it won't fail:
image

**Review by krdabrowski — APPROVED** [review]
Overall LGTM

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java (hunk 5 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java]: fixed

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java (hunk 4 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/messaging/in/mapper/RepoEventMapper.java]: fixed

**Pull request merged**